### PR TITLE
Ship new accessible file and symbol tree

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #12916

This ships the new file and symbol tree and removes the old implementation.

Two other small fixed I sneaked into this PR:

- We now use the `<Tabs>` behavior of `memoize` which means that when switching between tabs the state is kept alive (including scroll position and such). This also allows us to remove a layout effect and move it to be a passive effect in the files tree.
- Changed the microtask to a timeout to silence the React warning that flushSync can't happen in a lifecycle. Seems like React already uses microtasks for some batching.

## Test plan

- Tested it locally
- The new files and symbols tree have been tested extensively on S2 and dotcom
- I changed the existing tests to also work against the new files tree.

## App preview:

- [Web](https://sg-web-ps-rm-old-tree.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| -0.00% (-0.04 kb) | -0.18% (-27.04 kb) 🔽 | -0.23% (-27.01 kb) 🔽 | -0.26% (-2) 🔽 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/rm-old-tree/compare-report.html;tab=live_object) for a full comparison between the commits 64dbace and 639d5ff or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by philipp-spiess** [comment]
I feel like CI is getting closer to a green build so I’m opening this up for review. :D

**Review by taras-yemets — APPROVED** [review]
Great work, @philipp-spiess 💪🏻

**Review by valerybugakov — APPROVED** [review]
Yoohoo, so much red!

File: client/web/src/repo/RepoRevisionSidebarSymbols.test.tsx [hunk 1]
```diff
@@ -110,12 +110,17 @@ describe('RepoRevisionSidebarSymbols', () => {
...
+    it('clicking symbol updates route', async () => {
         expect(renderResult.locationRef.current?.search).toEqual('')
 
         const symbol = renderResult.getByText('firstSymbol')
         fireEvent.click(symbol)
 
+        // We need to synchronously flush inside the event handler and since this is warning in
+        // React 18, we've moved it to a setTimeout. This test needs to wait for this timeout to be
+        // flushed
+        await new Promise(resolve => setTimeout(resolve, 0))
```
Comment (valerybugakov) [inline client/web/src/repo/RepoRevisionSidebarSymbols.test.tsx:122]: This is so common that we have a utility for that which is frequently used:

```tsx
import delay from 'delay'

it('clicking symbol updates route', async () => {
  // ...
 
 await delay(0)

  // ...
})

**Pull request merged**